File: sn.infra.ui/lwp/lc.web.resources.base/eclipse/plugins/com.ibm.lconn.core.config/src/com/ibm/lconn/core/config/services/Services.java
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* OCO Source Materials                                              */
/*                                                                   */
/* Copyright IBM Corp. 2011, 2015                                    */
/*                                                                   */
/* The source code for this program is not published or otherwise    */
/* divested of its trade secrets, irrespective of what has been      */
/* deposited with the U.S. Copyright Office.                         */
/*                                                                   */
/* ***************************************************************** */

package com.ibm.lconn.core.config.services;

import java.net.URL;
import java.util.Collection;
import java.util.HashMap;
import java.util.Map;

import com.ibm.ventura.internal.config.api.VenturaConfigurationProvider;
import com.ibm.ventura.internal.config.exception.VenturaConfigException;
import com.ibm.ventura.internal.config.exception.VenturaConfigHelperException;
import com.ibm.ventura.internal.config.helper.api.VenturaConfigurationHelper;

public class Services {

	private static VenturaConfigurationProvider provider;
	public static String COMMON_PATH = null;
	public static String COMMON_PATH_SECURE = null;
	public static String CONTEXT_ROOT;

	private static Map<String, Object> services;
	private static final String WEBRESOURCES = "webresources";

	public static void init() {

		try {
			provider = VenturaConfigurationProvider.Factory.getInstance();
		} catch (VenturaConfigException e) {
			throw new VenturaConfigHelperException(e);
		}

		try {
			VenturaConfigurationHelper helper = VenturaConfigurationHelper.Factory
					.getInstance();
			Collection<String> serviceNames = helper.getInstalledComponents();

			services = new HashMap<String, Object>();
			for (String serviceName : serviceNames) {
				boolean isEnabled = provider.isServiceEnabled(serviceName);
				if (!isEnabled)
					continue;

				Map<String, Object> service = new HashMap<String, Object>();
				services.put(serviceName, service);
			}
			URL commonUrl = provider.getServiceURL(WEBRESOURCES);
			COMMON_PATH = commonUrl.toString();
			URL commonSecureUrl = provider.getSecureServiceURL(WEBRESOURCES);
			COMMON_PATH_SECURE = commonSecureUrl.toString();
			CONTEXT_ROOT = provider.getHrefPathPrefix(WEBRESOURCES);
		} catch (VenturaConfigHelperException e1) {
			throw new RuntimeException(e1);
		} catch (VenturaConfigException e2) {
			throw new RuntimeException(e2);
		}
	}
	
	// For runtime check in MT environment
	public static String getServiceUrl(boolean secure)
	{
		String ret = null;
		try {
			ret = secure ? provider.getSecureServiceURL(WEBRESOURCES).toString() : provider.getServiceURL(WEBRESOURCES).toString();
		} catch (VenturaConfigException e) {
			throw new RuntimeException(e);
		}
		return ret;
	}
}

File: sn.infra.ui/lwp/ic.web.resources/eclipse/plugins/com.ibm.ic.core.web.resources/resources/util/services.js
/* ***************************************************************** */
/*                                                                   */
/* IBM Confidential                                                  */
/*                                                                   */
/* OCO Source Materials                                              */
/*                                                                   */
/* Copyright IBM Corp. 2015, 2016                                    */
/*                                                                   */
/* The source code for this program is not published or otherwise    */
/* divested of its trade secrets, irrespective of what has been      */
/* deposited with the U.S. Copyright Office.                         */
/*                                                                   */
/* ***************************************************************** */

define([
        "dojo",
        "dojo/_base/declare",
        "dojo/request/xhr",
        "dojo/Deferred",
        "dojo/_base/lang"
], function(dj, djDeclare, xhr, djDeferred, lang) {
   
   var servicesUtil = {
      _serviceName: "",
      _userId: "",
      _orgId: "", 
      _services: {},
      
      isEnabled: function(serviceName, userId, orgId) {
         console.debug("Check service: " + serviceName + ", userId: " + userId + ", orgId: " + orgId);

         this._serviceName = serviceName || "";
         this._userId = userId || "";
         this._orgId = orgId || "";
         
         var deferred = new djDeferred();
         var dc = lconn.core.config.services.deploymentConfig; 
         var url = (((window.location.protocol||"http").replace(":","")=="https") ? dc.secureUrl : dc.url) + "/myserviceconfigs?format=json";   
         var key = "s:{" + this._serviceName + "},u{" + this._userId + "},o{" + this._orgId + "}";
         console.debug("services url is: " + url + " key is: " + key);
         
         var bindArgs={
            method: "GET",
            handleAs: "json",
            timeout: null,
            sync: false
         };
         
         if (this._services[key])
         {
            deferred.resolve((this._services[key] == "true")? true : false);
            console.debug("Service is found in _services object and the enabled value is " + this._services[key] );
            return deferred.promise;
         }
         
         that = this;
         xhr(url, bindArgs).then(
            function(response) {
               var enabled = that._handleResults(response, serviceName, key);  
               deferred.resolve(enabled);
               
            },
            function(response) {
               var enabled =that._handleError(response);
               deferred.resolve(enabled);
            }            
         ); 
         
         return deferred.promise;
      },
      
      _handleResults: function(data, serviceName, key) {
         console.debug("_handleResults: get response: " + JSON.stringify(data));
         var ret = data[serviceName] ? true : false;
         this._services[key] = ret? "true":"false";
         console.debug("_handleResults retVal is:" + ret);
         return ret;
      },
      
      _handleError: function(err) 
      {
        console.debug("Error when requesting json data" );         
         return false;
      }
      
   };
   
   lang.setObject("lconn.core.util.services", servicesUtil);
   
   return servicesUtil;
   
 })